test/integration/run_int_test.go

```diff
@@ -313,6 +314,43 @@ func (suite *RunIntegrationTestSuite) TestRun_Perl_Variable() {
 	cp.ExpectExitCode(0)
 }
 
+func (suite *RunIntegrationTestSuite) TestRun_Args() {
+	suite.OnlyRunForTags(tagsuite.Run)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	suite.createProjectFile(ts, 3)
+
+	asyFilename := filepath.Join(ts.Dirs.Work, "activestate.yaml")
+	asyFile, err := os.OpenFile(asyFilename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	suite.Require().NoError(err, "config is opened for appending")
+	defer asyFile.Close()
+
+	lang := project.DefaultScriptLanguage()[0].String()
+	cmd := `if [ "$1" = "<3" ]; then echo heart; fi`
+	if runtime.GOOS == "windows" {
+		cmd = `@echo off
+      if "%1"=="<3" (echo heart)` // need to match indent of YAML below
+	}
+	_, err = asyFile.WriteString(strings.TrimPrefix(fmt.Sprintf(`
+  - name: args
+    language: %s
+    value: |
+      %s
+`, lang, cmd), "\n"))
+	suite.Require().NoError(err, "extra config is appended")
+
+	arg := "<3"
+	if runtime.GOOS == "windows" {
+		// The '<' needs to be escaped with '^', and I don't know why. There is no way around it.
+		// The other exec and shell integration tests that test arg passing do not need this escape.
+		// Only this batch test does.
+		arg = "^<3"
```

Comment:
I feel like this is okay because if you are explicitly writing a batch script, then you ought to know the ins and outs of how batch handles incoming arguments, so if you're doing `state run <name> <args>` with funky `<args>`, then you ought to know what needs escaping and what does not.